internal/client/node_room_member.go

```diff
@@ -58,6 +58,152 @@ func (c *Client) GetAllNodes(spaceID string) (*RoomNodes, error) {
 	return roomNodes, nil
 }
 
+func (c *Client) ListNodeMembershipRules(spaceID, roomID string) (*[]NodeMembershipRule, error) {
```

Comment:
returning a pointer to a slice (`*[]NodeMembershipRule`) is unusual and generally not recommended in Go. For example:

- Slices are already reference types that contain a pointer to an underlying array
- Returning a pointer to a slice adds an unnecessary level of indirection
- It makes the caller deal with both nil slice and nil pointer cases

A more common approach would be to return `[]NodeMembershipRule` in this case